### ISSUE TITLE
Incorrect warning for world-writable configuration

### ISSUE THREAD

**Issue by topimiettinen** [systemd/systemd#12165]
**systemd version the issue has been seen with**
> 241

**Used distribution**
 > Debian Buster

**Expected behaviour you didn't see**
 > No message for symbolic links

**Unexpected behaviour you saw**
 > Log message:
`Configuration file /etc/systemd/system/systemd-resolved.service.d/resolvconf.conf is marked world-writable. Please remove world writability permission bits. Proceeding anyway.`

The referred path is a symbolic link to `/dev/null` and thus indeed may appear to be world-writable if only file permissions are checked, but of course it's not really so. 

**Steps to reproduce the problem**
Make a symbolic link to disable a distro drop-in:
```
$ ls -l /etc/systemd/system/systemd-resolved.service.d/resolvconf.conf /usr/lib/systemd/system/systemd-resolved.service.d/resolvconf.conf
lrwxrwxrwx  1 root root   9 Mar 20 12:57 /etc/systemd/system/systemd-resolved.service.d/resolvconf.conf -> /dev/null
-rw-r--r--. 1 root root 551 Feb 22 01:10 /usr/lib/systemd/system/systemd-resolved.service.d/resolvconf.conf
```
Link works as it should and disables the drop-in, as seen in log message:
`Skipping overridden file '/usr/lib/systemd/system/systemd-resolved.service.d/resolvconf.conf'.`

It would be easy to add a check to [src/shared/conf-parser.c](https://github.com/systemd/systemd/blob/76e62a4d2279a038b1e6feb95ecbfbbf9a57892d/src/shared/conf-parser.c#L302) to skip the permissions check `fd_warn_permissions()` (or adjust that function instead) in case of a symbolic link, but perhaps there's a deeper problem as this does not happen with other kinds of symbolic links used for masking? I don't get any other warnings of this kind for other links.

**Comment by poettering** [comment]
Fix waiting in #12177

**Comment by topimiettinen** [comment]
Nice, but why there's no error for other links, only just this case?
